### PR TITLE
Improve documentation on doc fragments

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Add information and examples on how to use additional properites from a doc fragment

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

`docs/docsite/rst/dev_guide/developing_modules_documenting.rst`

### CODE REVIEW

File: docs/docsite/rst/dev_guide/developing_modules_documenting.rst [hunk 1]
```diff
@@ -285,10 +285,41 @@ Documentation fragments
...
+
+    extends_documentation_fragment: example_fragment.other
+
+Or use both :
+
+.. code-block:: yaml+jinja
+
+    extends_documentation_fragment:
+      - example_fragment
+      - example_fragment.other
```
Comment (bcoca) [inline docs/docsite/rst/dev_guide/developing_modules_documenting.rst:322]: you should also note that they are 'overridable' to allow for things like adding `version_added` to the module even if the fragment was shared with preexisting modules.

Comment (samdoran) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst:322]: Meaning an imported fragment overrides an item defined in the module?

Comment (bcoca) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst:322]: the opposite, i.e fragment has `version_added: 1.9` module can have same option and just the overrideded key `version_added: "2.10"`

Comment (bcoca) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst:322]: the module is the 'more specific' in this case

Comment (samdoran) [reply docs/docsite/rst/dev_guide/developing_modules_documenting.rst:322]: Oh, yeah. That makes sense. I'll add that.

**Review by bcoca — APPROVED** [review]

**Pull request merged**